daemon/daemon.go

```diff
@@ -912,6 +912,10 @@ func NewDaemon(dp datapath.Datapath) (*Daemon, *endpointRestoreState, error) {
 	mtuConfig := mtu.NewConfiguration(option.Config.Tunnel != option.TunnelDisabled, option.Config.MTU)
 
 	if option.Config.EnableIPSec {
+		if len(option.Config.IPSecKeyFile) == 0 {
+			return nil, nil, fmt.Errorf("invalid %s option, ipsec key file is missing", option.EnableIPSecName)
```

Comment:
I think it would be more clear if instead of adding an explicit check here for an empty key file argument, the err checking of the result of `LoadIPSecKeysFile()` below formatted the error something like:

```
fmt.Errorf("failed to load IPSec Keys File %s: %v", option.Config.IPSecKeyFile, err)
```

Reply:
@joestringer Thanks! updated also added IPSecKeyFileName option